specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPackQueries_API.json

```diff
@@ -0,0 +1,464 @@
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/QueriesResource"
+            }
+          }
+        ],
+        "responses": {
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          },
+          "200": {
+            "description": "The new or updated query associated with the Log Analytics QueryPack.",
+            "schema": {
+              "$ref": "#/definitions/QueriesResource"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "QueryPut": {
+            "$ref": "./examples/QueryPackQueriesPut.json"
+          }
+        }
+      },
+      "delete": {
+        "description": "Deletes a specific Query defined within an Log Analytics QueryPack.",
+        "operationId": "Queries_Delete",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/QueryPackNameParameter"
+          },
+          {
+            "$ref": "#/parameters/QueryIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          },
+          "200": {
+            "description": "The query has been successfully removed from the Log Analytics QueryPack"
+          },
+          "204": {
+            "description": "The specified query or associated QueryPack does not exist."
+          }
+        },
+        "x-ms-examples": {
+          "QueryDelete": {
+            "$ref": "./examples/QueryPackQueriesDelete.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "ErrorResponse": {
+      "description": "Describe the format of an Error response.",
+      "type": "object",
+      "properties": {
+        "code": {
+          "description": "Error code",
+          "type": "string"
+        },
+        "message": {
+          "description": "Error message indicating why the operation failed.",
+          "type": "string"
+        }
+      }
+    },
+    "QueriesResource": {
+      "properties": {
+        "id": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Azure resource Id"
+        },
+        "properties": {
+          "$ref": "#/definitions/LogAnalyticsQueryPackQueryProperties"
+        }
```

Comment:
there is a bit of a circular reference going on here.  The "QueriesResource" contains QueryPackQueryProperties but the LogAnalyticsQueryPackQuery contains QueriesResource AND its own QueryPackQueryProperties

Reply:
yes youre right, fixed